crates/rethnet_eth/src/remote/client.rs

```diff
@@ -212,7 +291,24 @@ impl RpcClient {
 
     /// Calls `eth_chainId` and returns the chain ID.
     pub async fn chain_id(&self) -> Result<U256, RpcClientError> {
-        self.call(&MethodInvocation::ChainId()).await
+        // It's important to drop the read guard here, otherwise we'll deadlock when trying to
+        // acquire the write guard.
+        let chain_id = { *self.chain_id.read().await };
+
+        if let Some(chain_id) = chain_id {
+            Ok(chain_id)
+        } else {
+            // It's important to call without cache here to avoid infinite recursion as this method
```

Comment:
This made me realise that there are more methods that we'd like to call without caching, as not all methods are immutable. E.g. think about the RPC methods that return the server's version, etc.

Reply:
Thanks, will address in https://github.com/NomicFoundation/rethnet/issues/88